Created a data structure to contain reference particle information

cemitch99 opened the pull request:
Created a data structure "RefPart" in "ImpactXParticleContainer" to contain the reference particle information, and populated this in "initDist" using data from the input file.  The point of this PR is to validate the data structure.  If all is well, then RefPart values will be accessed or updated in individual elements.

cemitch99 committed:
Created a data structure to contain reference particle information
in "ImpactXParticleContainer", and populated this using values from input
file read by "initDist()".

ax3l commented:
Hi @cemitch99 , this looks great, thank you :sparkles: 

I'll add a small commit to propose just some aesthetic cleanups, let me know what you think (will push in a few minutes).

ax3l committed:
Reference Particle: Review

- Container API:
  - pass as struct, allowing easier addding/removing of
    attributes in the future
  - set default values and rename to "set" when setting
    the value (previously: "add")
- formatting: spacing, extra comments

cemitch99 reviewed (commented):
Great!  This looks cleaner, thanks.
